test/src/test/java/jenkins/slaves/OldRemotingAgentTest.java

```diff
@@ -0,0 +1,140 @@
+
+    @Rule
+    public TemporaryFolder tmpDir = new TemporaryFolder();
+
+    private File agentJar;
+
+    @Before
+    public void extractAgent() throws Exception {
+        agentJar = new File(tmpDir.getRoot(), "old-agent.jar");
+        FileUtils.copyURLToFile(getClass().getResource("/old-remoting/remoting-minimal-supported.jar"), agentJar);
+    }
+
+    @Test
+    @Issue("JENKINS-48761")
+    public void shouldBeAbleToConnectAgentWithMinimalSupportedVersion() throws Exception {
+        Label agentLabel = new LabelAtom("old-agent");
+        Slave agent = j.createOnlineSlave(agentLabel);
+        boolean isUnix = agent.getComputer().isUnix();
+        assertThat("Received wrong agent version. A minimal supported version is expected",
+                agent.getComputer().getSlaveVersion(),
+                equalTo(RemotingVersionInfo.getMinimalSupportedVersion().toString()));
+
+        // Just ensure we are able to run something on the agent
+        FreeStyleProject project = j.createFreeStyleProject("foo");
+        project.setAssignedLabel(agentLabel);
+        project.getBuildersList().add(isUnix ? new Shell("echo Hello") : new BatchFile("echo 'hello'"));
+        j.buildAndAssertSuccess(project);
+
+        // Run agent monitors
+        NodeMonitorAssert.assertMonitors(NodeMonitor.getAll(), agent.getComputer());
+    }
+
+    //TODO: move the logic to JTH
+    private class JenkinsRuleWithOldAgent extends JenkinsRule {
+
+        @Override
+        public ComputerLauncher createComputerLauncher(EnvVars env) throws URISyntaxException, IOException {
+
+            // EnvVars are ignored, simple Command Launcher does not offer this API in public
+            int sz = this.jenkins.getNodes().size();
+            return new SimpleCommandLauncher(String.format("\"%s/bin/java\" %s -jar \"%s\"",
+                    System.getProperty("java.home"),
+                    SLAVE_DEBUG_PORT > 0 ? " -Xdebug -Xrunjdwp:transport=dt_socket,server=y,address=" + (SLAVE_DEBUG_PORT + sz) : "",
+                    agentJar.getAbsolutePath()));
+        }
+    }
+
+    private static class NodeMonitorAssert extends NodeMonitor {
+
+        static void assertMonitors(Collection<NodeMonitor> toCheck, Computer c) throws AssertionError {
+            for (NodeMonitor monitor : toCheck) {
+                assertMonitor(monitor, c);
+            }
+        }
+
+        static void assertMonitor(NodeMonitor monitor, Computer c) throws AssertionError {
+            AbstractNodeMonitorDescriptor<?> descriptor = monitor.getDescriptor();
+            final Method monitorMethod;
+            try {
+                monitorMethod = AbstractAsyncNodeMonitorDescriptor.class.getDeclaredMethod("monitor", Computer.class);
+            } catch (NoSuchMethodException ex) {
+                System.out.println("Cannot invoke monitor " + monitor + ", no monitor(Computer.class) method in the Descriptor. It will be ignored. " + ex.getMessage());
```

Comment:
This is test code. Just throw it up!

Or, better, make that method `@Restricted(NoExternalUse.class) public`.

Reply:
Right, it should never happen